### PR TITLE
Add handling for WKT as extent

### PR DESCRIPTION
**Pull request by JohannaOtt**
WGS-918 / SVC-750

@stempler Do we need to take anything specific into account with the use of the snippet and the AdV using the gradle setup?

### CODE REVIEW

File: annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex [hunk 1]
```diff
@@ -1,21 +1,23 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<hale-project version="4.0.0.SNAPSHOT">
+<hale-project version="4.1.0.SNAPSHOT">
     <name>[Annex III] AdV ALKIS zu INSPIRE Existing Land Use</name>
     <author>Johanna Ott</author>
     <description>Transformationsprojekt für die Transformation von Daten aus ALKIS nach INSPIRE ExistingLandUse.&#xD;
 &#xD;
-ATKIS als Quelle derzeit ausgeschlossen&#xD;
-(Begründung aus Email Herr Schliebner 30.05.2018: "Keine Verflächung von linienhaften Elementen. Zum AdV-Plenums-Beschluss 127/7 soll ein entsprechendes Änderungsersuchen gestellt werden, was Herr Dr. Seifert am 05.-06.Juni 2018 bei der AK IK-Tagung thematisieren wird. Diese Änderung kann hieb- und stichfest damit begründet werden, dass zur Schaffung von INSPIRE-Daten aus ATKIS erheblicher Aufwand anfiele.")&#xD;
+Hinweis zur Angabe einer WKT-Geometrie des Bundeslandes zur korrekten HAnhabung von ExistingLandUseDataSet.extent:&#xD;
```
Comment (florianesser) [inline annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex:7]: *Handhabung

File: annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+	Mit der Transformation wird die Datei "extentBundesland.groovy" bereitgestellt. Um den INSPIRE-Datensatz vollständig korrekt zu erzeugen, muss dort der Platzhalter [..] mit einer WKT-Geometrie Ihres Bundeslandes ersetzt werden (die Anführungszeichen nicht ersetzen). Machen Sie dort keine Angabe, wird die konvexe Hülle der Ausgangsdaten dort verwendet, was aber nicht den INSPIRE-Anforderungen entspricht.&#xD;
```
Comment (florianesser) [inline annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex:8]: I tried that with a real-world example (Landesgrenze Hessen), and it failed with the message below. I think we have to think of an alternative approach, as I assume the WKT string of most Landesgrenzen will exceed the limit.

image

**Review by florianesser — CHANGES_REQUESTED** [review]

File: annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex (hunk 1 above)
Comment (JohannaOtt) [reply annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex:7]: Good catch, thanks!

File: annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex (hunk 2 above)
Comment (JohannaOtt) [reply annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex:8]: That is an issue indeed. I only tested with smaller geometries.
What about asking them to provide the boundaries in the source foulder in an xml file using an xsd defined by us and read it as additional source data?
In this case we could again provide the xml file and they would only need to fill in the geometry. What do you think?

Comment (JohannaOtt) [reply annex-2-3/mappings/LandUse/ALKIS2INSPIRE_ELU.halex]: I implemented a different approach using a csv file.

**Comment by JohannaOtt** [comment]
@florianesser We would also need to adapt the gradle handling in order to make sure that the file "extentBundesland.csv" is included in every run of the ELU alignment, right?

**Comment by florianesser** [comment]
> @florianesser We would also need to adapt the gradle handling in order to make sure that the file "extentBundesland.csv" is included in every run of the ELU alignment, right?

Yes, I've added support for that to `build.gradle`

**Comment by JohannaOtt** [comment]
@florianesser Should we move the csv schema to a URL? Currently, the local path on my machine is still included in the halex files.

**Review by florianesser — APPROVED** [review]
LGTM

**Pull request merged**